app/core/services/document_processor.py

```
@@ -0,0 +1,220 @@
+    if failed_reason:
+        payload["failed_reason"] = failed_reason
+
+    try:
+        async with httpx.AsyncClient() as http:
+            resp = await http.patch(
+                f"{CORE_BASE_URL}/internal/v1/sites/{site_id}/documents/{doc_id}/status",
+                json=payload,
+                timeout=10.0,
+            )
+            resp.raise_for_status()
+    except Exception as e:
```

Ruff (0.15.9)

[warning] Do not catch blind exception: `Exception`

(BLE001)